troposphere/awslambda.py

```diff
@@ -38,6 +38,10 @@ def validate(self):
             raise ValueError(
                 "You can't specify both 'S3ObjectVersion' and 'ZipFile'"
             )
+        if zip_file and len(zip_file) > 4096:
```

Comment:
You are correct. In this case there is not way to check it locally as there could be any Ref, or GetAttr etc. So what if we run this check on strings only?

Maybe also on simple cases like Join('', [ ...]) where all elements are strings and we can actually calculate the size. What do you think?